### PR TITLE
Add safari technology preview for babel-preset-env.

### LINKED ISSUE #6608: Safari TP in browserslist causes exception
> Issue originally reported by @bhollis in babel/babel-preset-env#393

If I specify "unreleased versions" in browserslist, it includes "Safari TP", which makes babel-preset-env angry:

```
Error: Invalid version passed for target "safari": "TP.0.0". Versions must be in semver format (major.minor.patch) (While processing preset: "/Users/brh/Documents/oss/DIM/node_modules/babel-preset-env/lib/index.js")
    at /Users/brh/Documents/oss/DIM/node_modules/babel-preset-env/lib/index.js:69:13
```

### PR DESCRIPTION
**Pull request by yavorsky**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #6608, https://github.com/babel/babel-preset-env/issues/393
| Patch: Bug Fix?          | n
| Major: Breaking Change?  | n
| Minor: New Feature?      | y
| Tests Added + Pass?      | Yes
| Documentation PR         | y
| Any Dependency Changes?  | n
| License                  | MIT

Original PR (https://github.com/babel/babel-preset-env/pull/402)
was modified according to the `shippedProposals` option.
Added fixtures for *proposal-object-rest-spread* and *proposal-optional-catch-binding*, which shouldn't transpile code for safari TP.

### CODE REVIEW

File: experimental/babel-preset-env/README.md [hunk 1]
```diff
@@ -181,7 +181,7 @@ For more information on setting options for a preset, refer to the [plugin/prese
 
 Takes an object of environment versions to support.
 
-Each target environment takes a number or a string (we recommend using a string when specifying minor versions like `node: "6.10"`).
+Each target environment takes a number or a string (we recommend using a string when specifying minor versions like `node: "6.10"`). You can also specify `tp` (technology preview) version for Safari.
```
Comment (hzoo) [inline experimental/babel-preset-env/README.md]: I don't think we need a whole line of documentation just for safari tp? or I would just name the value `safari tp`? below

Comment (yavorsky) [reply experimental/babel-preset-env/README.md]: @hzoo According to the compat-table it's pretty static value.
All chrome canary releases are labeled under number value. For example, chrome 64 (canary).
Same for firefox: FF 58 (Nightly).
Edge 17 (Preview).
But for Safari it's just TP. And when new TP become 11.2 for example, the new one will be labeled as tp also.

File: experimental/babel-preset-env/scripts/build-data.js [hunk 2]
```diff
@@ -185,24 +186,32 @@ const getLowestImplementedVersion = ({ features }, env) => {
...
     // don't count against it
     // only doing this for built-ins atm
     if (!test.babel && isBuiltIn) {
       return "-1";
     }
+    const sortedKeys = Object.keys(test);
 
     return (
-      Object.keys(test)
+      sortedKeys
```
Comment (existentialism) [inline experimental/babel-preset-env/scripts/build-data.js]: Nit: we can revert this change too
